src/models/forwardUrl.ts

```diff
@@ -82,3 +82,27 @@ const prune = async (
     [endpointId, userId, limit],
   );
 };
+
+export const deleteForwardUrls = async (
+  userId: number,
+  endpointId: number,
+  forwardUrl: string,
+) => {
+  const { rowCount } = await query(
+    `
+  DELETE FROM forward_urls as f
+  WHERE
+    f.endpoint_id = $1
+    AND
+    f.url = $2
+    AND
+    EXISTS (
+      SELECT 1
+      FROM endpoints as e
+      INNER JOIN user_endpoints as ue on ue.endpoint_id = e.id
+      WHERE e.id = f.endpoint_id AND ue.user_id = $3
+    )`,
```

Comment:
This `EXISTS` seems unnecessarily complex. I believe you got inspiration from here: https://github.com/webhook-buddy/webhook-buddy-server/blob/master/src/models/webhook.ts#L167-L172

But in that sample code, the extra EXISTS is an extra layer of protection to ensure that someone isn't deleting a webshook to which s/he doesn't have access. In your case, the user_id is in the forward_urls table, so we can just check for that. So something like this should do the trick:

```
AND
f.user_id = $3
```

Reply:
Thanks for clarifying! I have corrected and pushed. Tested in graphql playground and whb-client, it still works.

Reply:
@johnnyoshika